doc/guides/platformio/index.rst

```diff
@@ -0,0 +1,91 @@
+.. _platformio:
+
+Using with PlatformIO
+#####################
+
+.. image:: img/platformio-logo.png
+  :target: https://platformio.org/?utm_source=docs.zephyrproject.org
+
+.. contents::
+    :local:
+
+What is PlatformIO?
+*******************
+
+`PlatformIO <https://platformio.org/?utm_source=docs.zephyrproject.org>`__
```

Comment:
I don't think this section belongs in Zephyr's documentation.  This should be part of PlaformIO's doc, and can be linked from here.

Reply:
Most of this seems to be links to other places in the PlatformIO documentation, no? I'm not sure we've ever handled an integration like this and what sorts of details should go here.

In general, this page reads a bit too much like an advertisement for PlatformIO. I'm sure that it's great, I just don't think things like this belong here:

- "professional"
- "new generation toolset"
- "unique, developed-from-scratch build system that removes the usual pain"
- "easy integration with numerous cloud platforms"

I think this documentation should just stick to the facts if it belongs here at all. Will leave that up to others to decide.